Solutions/Corvus.Json.Specs/Features/JsonPatch/builder_spec_tests.feature

```diff
@@ -0,0 +1,82 @@
+
+Feature: builder_spec_tests
+
+Scenario: 4.1. add with missing object [e9311282-4ed5-4580-85bc-e9b3e3d09814]
+	Given the document {"q":{"bar":2}}
+	When I build the patch [{"op":"add","path":"/a/b","value":1}]
+	Then a patch exception should be thrown
+
+Scenario: A.1.  Adding an Object Member [c048fa20-b3d8-4aa3-bd44-077db15733e4]
+	Given the document {"foo":"bar"}
+	When I build the patch [{"op":"add","path":"/baz","value":"qux"}]
+	Then the patch result should equal {"baz":"qux","foo":"bar"}
+
+Scenario: A.2.  Adding an Array Element [c4026a32-8b55-42b4-a466-81719b076125]
+	Given the document {"foo":["bar","baz"]}
+	When I build the patch [{"op":"add","path":"/foo/1","value":"qux"}]
+	Then the patch result should equal {"foo":["bar","qux","baz"]}
+
+Scenario: A.3.  Removing an Object Member [61493012-a08e-4dab-9831-8cc64e37e571]
+	Given the document {"baz":"qux","foo":"bar"}
+	When I build the patch [{"op":"remove","path":"/baz"}]
+	Then the patch result should equal {"foo":"bar"}
+
+Scenario: A.4.  Removing an Array Element [ab4e5a07-0a4c-4027-ad49-71513a886102]
+	Given the document {"foo":["bar","qux","baz"]}
+	When I build the patch [{"op":"remove","path":"/foo/1"}]
+	Then the patch result should equal {"foo":["bar","baz"]}
+
+Scenario: A.5.  Replacing a Value [32231887-dcc1-4ea3-9fbf-90e2801b6c71]
+	Given the document {"baz":"qux","foo":"bar"}
+	When I build the patch [{"op":"replace","path":"/baz","value":"boo"}]
+	Then the patch result should equal {"baz":"boo","foo":"bar"}
+
+Scenario: A.6.  Moving a Value [5a770e0f-1221-4d08-af07-94f57ae60e31]
+	Given the document {"foo":{"bar":"baz","waldo":"fred"},"qux":{"corge":"grault"}}
+	When I build the patch [{"op":"move","from":"/foo/waldo","path":"/qux/thud"}]
+	Then the patch result should equal {"foo":{"bar":"baz"},"qux":{"corge":"grault","thud":"fred"}}
+
+Scenario: A.7.  Moving an Array Element [e905c322-0663-4d90-a304-d7304fa542c3]
+	Given the document {"foo":["all","grass","cows","eat"]}
+	When I build the patch [{"op":"move","from":"/foo/1","path":"/foo/3"}]
+	Then the patch result should equal {"foo":["all","cows","eat","grass"]}
+
+Scenario: A.8.  Testing a Value: Success [bf9eb3c7-a173-450f-ae90-fec054e12672]
+	Given the document {"baz":"qux","foo":["a",2,"c"]}
+	When I build the patch [{"op":"test","path":"/baz","value":"qux"},{"op":"test","path":"/foo/1","value":2}]
+	Then the patch result should equal {"baz":"qux","foo":["a",2,"c"]}
```

Comment:
I have added an additional test to expressly validate that the patch result was set, rather than an exception thrown.

(An exception would be thrown because the patch is not available in the failure case you identify, but this just results in the test failing because of a null reference which is correct, but not very clear!)